frontend/src/common/TagSelector.tsx

```diff
@@ -0,0 +1,96 @@
+import Button from '@mui/material/Button'
+import Chip from '@mui/material/Chip'
+import Typography from '@mui/material/Typography'
+import { ReactElement, useEffect, useState } from 'react'
+
+type PartialTagSelectorProps =
+  | {
+      multiple: true
+      onChange: (value: string[]) => void
+    }
+  | {
+      multiple?: false
+      onChange: (value: string) => void
+    }
+
+type TagSelectorProps = {
+  label: string
+  tags: string[]
+  size?: 'small' | 'medium'
+  expandThreshold?: number
+} & PartialTagSelectorProps
+
+export default function TagSelector({
+  label,
+  tags,
+  onChange,
+  multiple,
+  size = 'medium',
+  expandThreshold = 5,
+}: TagSelectorProps): ReactElement {
+  const [selectedTags, setSelectedTags] = useState<string[]>([])
+  const [expanded, setExpanded] = useState(false)
+
+  useEffect(() => {
+    if (multiple) {
+      onChange(selectedTags)
+    } else {
+      onChange(selectedTags[0])
+    }
+  }, [selectedTags, multiple, onChange])
+
+  const handleChange = (selectedTag: string): void => {
+    if (multiple) {
+      if (selectedTags.includes(selectedTag)) {
+        setSelectedTags(selectedTags.filter((tag) => tag !== selectedTag))
+      } else {
+        setSelectedTags([...selectedTags, selectedTag])
+      }
+    } else {
+      setSelectedTags(selectedTags[0] !== selectedTag ? [selectedTag] : [])
+    }
+  }
+
+  function toggleExpansion(): void {
+    setExpanded(!expanded)
+  }
+
+  const allTags = tags.map((tag) => (
+    <Tag key={tag} tag={tag} size={size} active={selectedTags.includes(tag)} handleChange={handleChange} />
+  ))
+
+  function displaySelectedTagCount(): string {
+    return multiple && selectedTags.length > 0 ? `(${selectedTags.length} selected)` : ''
+  }
+
+  return (
+    <>
+      <Typography variant='h6'>{`${label} ${displaySelectedTagCount()}`}</Typography>
+      {!expanded && allTags.slice(0, expandThreshold)}
+      {expanded && allTags}
+      {tags.length > expandThreshold && (
+        <Button onClick={toggleExpansion}>{expanded ? 'Show less' : 'Show more...'}</Button>
+      )}
+    </>
+  )
+}
+
+type TagProps = {
+  tag: string
+  handleChange: (value: string) => void
+  size?: TagSelectorProps['size']
+  active: boolean
+}
+
+function Tag({ tag, handleChange, size, active }: TagProps) {
```

Comment:
Same here, it looks like this is likely to represent more than just tags. Maybe rename to `SelectableChip` or `ActiveChip`?